src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -428,7 +342,225 @@ private Expression ProjectElement(Expression source, SelectExpandClause selectEx
             return Expression.MemberInit(Expression.New(wrapperType), wrapperTypeMemberAssignments);
         }
 
-        private static bool GetSelectsOpenTypeSegments(SelectExpandClause selectExpandClause, IEdmStructuredType structuredType)
+        /// <summary>
+        /// Gets the $select and $expand properties from the given <see cref="SelectExpandClause"/>
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="structuredType">The current structural type.</param>
+        /// <param name="navigationSource">The current navigation source.</param>
+        /// <param name="selectExpandClause">The given select and expand clause.</param>
+        /// <param name="propertiesToInclude">The out properties to include at current level, could be null.</param>
+        /// <param name="propertiesToExpand">The out properties to expand at current level, could be null.</param>
+        /// <param name="autoSelectedProperties">The out auto selected properties to include at current level, could be null.</param>
+        /// <returns>true if the select contains dynamic property selection, false if it's not.</returns>
```

Comment:
>true if the select contains dynamic property selection [](start = 21, length = 54)

what do you mean by "contains dynamic property selection"? at first I thought this had to do with dynamic properties, but I think it just means there is an explicit property selection.  If the later, we should rename the variable and update the description to avoid confusion.

Reply:
This method travels all select and expand and records "which property is included", "Which property is expanded", meanwhile, if it includes a dynamic property selection, return "true". 

Which "variable" do you think to rename? 


---
In reply to: [355537085](https://github.com/OData/WebApi/pull/1941#discussion_r355537085) [](ancestors = 355537085)

Reply:
isContainDynamicPropertySelect is confusing; maybe selectContainsDynamicProperty?

---
In reply to: [355649755](https://github.com/OData/WebApi/pull/1941#discussion_r355649755) [](ancestors = 355649755,355537085)